<div style="display: flex; background-color: RGB(100,149,237);" >
<h1 style="margin: auto; padding: 30px; ">Mission 1 </h1>
</div>

<div style="display: flex; background-color: RGB(100,149,237);" >
<h3 style="margin: auto; padding: 30px; ">Importation et visualisation des données </h3>
</div>

In [11]:
import pandas as pd

In [12]:
customers = pd.read_csv("data/olist_customers_dataset.csv")
geolocation = pd.read_csv("data/olist_geolocation_dataset.csv")
items = pd.read_csv("data/olist_order_items_dataset.csv")
payments = pd.read_csv("data/olist_order_payments_dataset.csv")
reviews = pd.read_csv("data/olist_order_reviews_dataset.csv")
orders = pd.read_csv("data/olist_orders_dataset.csv")
products = pd.read_csv("data/olist_products_dataset.csv")
sellers = pd.read_csv("data/olist_sellers_dataset.csv")
translate = pd.read_csv("data/product_category_name_translation.csv")

In [13]:
translate.head()

product_category_name product_category_name_english
0            beleza_saude                 health_beauty
1  informatica_acessorios         computers_accessories
2              automotivo                          auto
3         cama_mesa_banho                bed_bath_table
4        moveis_decoracao               furniture_decor

In [14]:
translate_mapper = dict(zip(translate.product_category_name, translate.product_category_name_english))

In [15]:
products['product_category_name'] = products.product_category_name.map(translate_mapper)

In [16]:
products.product_category_name.value_counts()

bed_bath_table               3029
sports_leisure               2867
furniture_decor              2657
health_beauty                2444
housewares                   2335
                             ... 
tablets_printing_image          9
fashion_childrens_clothes       5
home_comfort_2                  5
security_and_services           2
cds_dvds_musicals               1
Name: product_category_name, Length: 71, dtype: int64

In [17]:
customers.customer_unique_id.value_counts()

8d50f5eadf50201ccdcedfb9e2ac8455    17
3e43e6105506432c953e165fb2acf44c     9
1b6c7548a2a1f9037c1fd3ddfed95f33     7
ca77025e7201e3b30c44b472ff346268     7
6469f99c1f9dfae7733b25662e7f1782     7
                                    ..
081f07439678af2da7755a0aa572154d     1
9c57789a6a587f4efcfb01e36c106014     1
9e4c21635f640562de1a6976374fba03     1
06c0ea6ee892364d1608ee47aa9f56a0     1
84732c5050c01db9b23e19ba39899398     1
Name: customer_unique_id, Length: 96096, dtype: int64

<div style="display: flex; background-color: RGB(100,149,237);" >
<h3 style="margin: auto; padding: 30px; ">Frequence </h3>
</div>

**Nous calculons la fréquence de commande d'un client.**

In [20]:
frequence = pd.merge(customers , orders, on = 'customer_id' , how = 'inner')
frequence

customer_id                customer_unique_id  \
0      06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1      18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2      4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3      b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4      4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   
...                                 ...                               ...   
99436  17ddf5dd5d51696bb3d7c6291687be6f  1a29b476fee25c95fbafc67c5ac95cf8   
99437  e7b71a9017aa05c9a7fd292d714858e8  d52a67c98be1cf6a5c84435bd38d095d   
99438  5e28dfe12db7fb50a4b2f691faecea5e  e9f50caf99f032f0bf3c55141f019d99   
99439  56b18e2166679b8a959d72dd06da27f9  73c2643a0a458b49f58cea58833b192e   
99440  274fa6071e5e17fe303b9748641082c8  84732c5050c01db9b23e19ba39899398   

       customer_zip_code_prefix          customer_city customer_state  \
0                         14409                 franca             SP   
1                          9790  sao bernardo do campo             SP   
2                          1151              sao paulo             SP   
3                          8775        mogi das cruzes             SP   
4                         13056               campinas             SP   
...                         ...                    ...            ...   
99436                      3937              sao paulo             SP   
99437                      6764        taboao da serra             SP   
99438                     60115              fortaleza             CE   
99439                     92120                 canoas             RS   
99440                      6703                  cotia             SP   

                               order_id order_status order_purchase_timestamp  \
0      00e7ee1b050b8499577073aeb2a297a1    delivered      2017-05-16 15:05:35   
1      29150127e6685892b6eab3eec79f59c7    delivered      2018-01-12 20:48:24   
2      b2059ed67ce144a36e2aa97d2c9e9ad2    delivered      2018-05-19 16:07:45   
3      951670f92359f4fe4a63112aa7306eba    delivered      2018-03-13 16:06:38   
4      6b7d50bd145f6fc7f33cebabd7e49d0f    delivered      2018-07-29 09:51:30   
...                                 ...          ...                      ...   
99436  6760e20addcf0121e9d58f2f1ff14298    delivered      2018-04-07 15:48:17   
99437  9ec0c8947d973db4f4e8dcf1fbfa8f1b    delivered      2018-04-04 08:20:22   
99438  fed4434add09a6f332ea398efd656a5c    delivered      2018-04-08 20:11:50   
99439  e31ec91cea1ecf97797787471f98a8c2    delivered      2017-11-03 21:08:33   
99440  28db69209a75e59f20ccbb5c36a20b90    delivered      2017-12-19 14:27:23   

         order_approved_at order_delivered_carrier_date  \
0      2017-05-16 15:22:12          2017-05-23 10:47:57   
1      2018-01-12 20:58:32          2018-01-15 17:14:59   
2      2018-05-20 16:19:10          2018-06-11 14:31:00   
3      2018-03-13 17:29:19          2018-03-27 23:22:42   
4      2018-07-29 10:10:09          2018-07-30 15:16:00   
...                    ...                          ...   
99436  2018-04-07 16:08:45          2018-04-11 02:08:36   
99437  2018-04-04 08:35:12          2018-04-05 18:42:35   
99438  2018-04-08 20:30:03          2018-04-09 17:52:17   
99439  2017-11-03 21:31:20          2017-11-06 18:24:41   
99440  2017-12-19 18:50:39          2017-12-21 19:17:21   

      order_delivered_customer_date order_estimated_delivery_date  
0               2017-05-25 10:35:35           2017-06-05 00:00:00  
1               2018-01-29 12:41:19           2018-02-06 00:00:00  
2               2018-06-14 17:58:51           2018-06-13 00:00:00  
3               2018-03-28 16:04:25           2018-04-10 00:00:00  
4               2018-08-09 20:55:48           2018-08-15 00:00:00  
...                             ...                           ...  
99436           2018-04-13 20:06:37           2018-04-25 00:00:00  


In [25]:
# converting date columns to datetime
date_columns = ['order_purchase_timestamp', 'order_approved_at', 'order_delivered_carrier_date', 'order_delivered_customer_date', 'order_estimated_delivery_date']
for col in date_columns:
    frequence[col] = pd.to_datetime(frequence[col], format='%Y-%m-%d %H:%M:%S')

In [26]:
frequence.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 99441 entries, 0 to 99440
Data columns (total 12 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0   customer_id                    99441 non-null  object        
 1   customer_unique_id             99441 non-null  object        
 2   customer_zip_code_prefix       99441 non-null  int64         
 3   customer_city                  99441 non-null  object        
 4   customer_state                 99441 non-null  object        
 5   order_id                       99441 non-null  object        
 6   order_status                   99441 non-null  object        
 7   order_purchase_timestamp       99441 non-null  datetime64[ns]
 8   order_approved_at              99281 non-null  datetime64[ns]
 9   order_delivered_carrier_date   97658 non-null  datetime64[ns]
 10  order_delivered_customer_date  96476 non-null  datetime64[ns]
 11  order_estimated

In [30]:
#frequence['frequence'] = frequence['customers_id'] frequence['order_purchase_timestamp']
df_frequency = frequence.groupby('customer_unique_id').order_purchase_timestamp.count().reset_index()
df_frequency.columns = ['customer_unique_id','order_purchase_timestamp']

#add this data to our main dataframe
df_frequency.describe()

order_purchase_timestamp
count              96096.000000
mean                   1.034809
std                    0.214384
min                    1.000000
25%                    1.000000
50%                    1.000000
75%                    1.000000
max                   17.000000

**il y a au total 96096 clients.</br>
la moyennes est de 1.034.</br>
écart-type est de 0.214.</br>
la mediane nous montre que les clients ne commande qu'une seule fois.</br>
75 pourcent des clients ne commande que 1 seule fois.</br>
et au maximum il y a 17 commandes du meme client.</br>**

<div style="display: flex; background-color: RGB(100,149,237);" >
<h3 style="margin: auto; padding: 30px; ">Panier Moyen </h3>
</div>

**Panier moyen par client**

In [42]:
panier = pd.merge(frequence , items, on = 'order_id' , how = 'inner')
panier

customer_id                customer_unique_id  \
0       06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1       18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2       4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3       b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4       4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   
...                                  ...                               ...   
112645  17ddf5dd5d51696bb3d7c6291687be6f  1a29b476fee25c95fbafc67c5ac95cf8   
112646  e7b71a9017aa05c9a7fd292d714858e8  d52a67c98be1cf6a5c84435bd38d095d   
112647  5e28dfe12db7fb50a4b2f691faecea5e  e9f50caf99f032f0bf3c55141f019d99   
112648  56b18e2166679b8a959d72dd06da27f9  73c2643a0a458b49f58cea58833b192e   
112649  274fa6071e5e17fe303b9748641082c8  84732c5050c01db9b23e19ba39899398   

        customer_zip_code_prefix          customer_city customer_state  \
0                          14409                 franca             SP   
1                           9790  sao bernardo do campo             SP   
2                           1151              sao paulo             SP   
3                           8775        mogi das cruzes             SP   
4                          13056               campinas             SP   
...                          ...                    ...            ...   
112645                      3937              sao paulo             SP   
112646                      6764        taboao da serra             SP   
112647                     60115              fortaleza             CE   
112648                     92120                 canoas             RS   
112649                      6703                  cotia             SP   

                                order_id order_status  \
0       00e7ee1b050b8499577073aeb2a297a1    delivered   
1       29150127e6685892b6eab3eec79f59c7    delivered   
2       b2059ed67ce144a36e2aa97d2c9e9ad2    delivered   
3       951670f92359f4fe4a63112aa7306eba    delivered   
4       6b7d50bd145f6fc7f33cebabd7e49d0f    delivered   
...                                  ...          ...   
112645  6760e20addcf0121e9d58f2f1ff14298    delivered   
112646  9ec0c8947d973db4f4e8dcf1fbfa8f1b    delivered   
112647  fed4434add09a6f332ea398efd656a5c    delivered   
112648  e31ec91cea1ecf97797787471f98a8c2    delivered   
112649  28db69209a75e59f20ccbb5c36a20b90    delivered   

       order_purchase_timestamp   order_approved_at  \
0           2017-05-16 15:05:35 2017-05-16 15:22:12   
1           2018-01-12 20:48:24 2018-01-12 20:58:32   
2           2018-05-19 16:07:45 2018-05-20 16:19:10   
3           2018-03-13 16:06:38 2018-03-13 17:29:19   
4           2018-07-29 09:51:30 2018-07-29 10:10:09   
...                         ...                 ...   
112645      2018-04-07 15:48:17 2018-04-07 16:08:45   
112646      2018-04-04 08:20:22 2018-04-04 08:35:12   
112647      2018-04-08 20:11:50 2018-04-08 20:30:03   
112648      2017-11-03 21:08:33 2017-11-03 21:31:20   
112649      2017-12-19 14:27:23 2017-12-19 18:50:39   

       order_delivered_carrier_date order_delivered_customer_date  \
0               2017-05-23 10:47:57           2017-05-25 10:35:35   
1               2018-01-15 17:14:59           2018-01-29 12:41:19   
2               2018-06-11 14:31:00           2018-06-14 17:58:51   
3               2018-03-27 23:22:42           2018-03-28 16:04:25   
4               2018-07-30 15:16:00           2018-08-09 20:55:48   
...                             ...                           ...   
112645          2018-04-11 02:08:36           2018-04-13 20:06:37   
112646          2018-04-05 18:42:35           2018-04-11 18:54:45   
112647          2018-04-09 17:52:17           2018-05-09 19:03:15   
112648          2017-11-06 18:24:41           2017-11-16 19:58:39   
112649          2017-12-21 19:17:21           2017-12-26 18:42:36   

       order_estimated_deliv

In [44]:
panier.order_id.value_counts()

8272b63d03f5f79c56e9e4120aec44ef    21
ab14fdcfbe524636d65ee38360e22ce8    20
1b15974a0141d54e36626dca3fdc731a    20
428a2f660dc84138d969ccd69a0ab6d5    15
9ef13efd6949e4573a18964dd1bbe7f5    15
                                    ..
25d25cce0abe624c6ab42ad5e9e30820     1
5a87769ff430e7d1873e2bb9b3968af3     1
ff5bf12997a3e2a32981ad70369bd708     1
da56812895f62c346de61efef3c1787b     1
28db69209a75e59f20ccbb5c36a20b90     1
Name: order_id, Length: 98666, dtype: int64

In [46]:
panier.order_id.value_counts().mean()

1.1417306873695092

In [48]:
panier.order_id.value_counts().describe()

count    98666.000000
mean         1.141731
std          0.538452
min          1.000000
25%          1.000000
50%          1.000000
75%          1.000000
max         21.000000
Name: order_id, dtype: float64

In [59]:
df_panier = panier.groupby("customer_id")['price'].mean().reset_index()

#add this data to our main dataframe
df_panier

customer_id   price
0      00012a2ce6f8dcda20d059ce98491703   89.80
1      000161a058600d5901f007fab4c27140   54.90
2      0001fd6190edaaf884bcaf3d49edf079  179.99
3      0002414f95344307404f0ace7a26f1d5  149.90
4      000379cdec625522490c315e70c7a9fb   93.00
...                                 ...     ...
98661  fffcb937e9dd47a13f05ecb8290f4d3e   78.00
98662  fffecc9f79fd8c764f843e9951b11341   54.90
98663  fffeda5b6d849fbd39689bb92087f431   47.90
98664  ffff42319e9b2d713724ae527742af25  199.90
98665  ffffa3172527f765de70084a7e53aae8   10.90

[98666 rows x 2 columns]

In [69]:
panier[panier.price == 'ffffa3172527f765de70084a7e53aae8']

customer_id                customer_unique_id  \
15307  c6e2731c5b391845f6800c97401a43a9  dc4802a71eae9be1dd28f5d788ceb526   

       customer_zip_code_prefix customer_city customer_state  \
15307                     79042  campo grande             MS   

                               order_id order_status order_purchase_timestamp  \
15307  0812eb902a67711a1cb742b3cdaa65ae    delivered      2017-02-12 20:37:36   

        order_approved_at order_delivered_carrier_date  \
15307 2017-02-12 20:45:12          2017-02-16 09:23:13   

      order_delivered_customer_date order_estimated_delivery_date  \
15307           2017-03-03 14:23:18                    2017-03-09   

       order_item_id                        product_id  \
15307              1  489ae2aa008f021502940f251d4cce7f   

                              seller_id  shipping_limit_date   price  \
15307  e3b4998c7a498169dc7bce44e6bb6277  2017-02-16 20:37:36  6735.0   

       freight_value  
15307         194.31

In [60]:
df_panier.describe()

price
count  98666.000000
mean     125.919255
std      190.985636
min        0.850000
25%       41.990000
50%       79.000000
75%      139.900000
max     6735.000000

In [ ]:
**il y a au total 98666 clients.</br>
la moyennes est de 125.91 Dollars </br>
écart-type est de 190.98.</br>
la mediane nous montre que les clients depense 79 Dollars.</br>
75 pourcent des clients ne commande que 1 seule fois.</br>
et au maximum il y a 17 commandes du meme client.</br>**